File: 2154-keep-multiplying-found-values-by-two/2154-keep-multiplying-found-values-by-two.py
class Solution(object):
    def findFinalValue(self, nums, original):
        """
        :type nums: List[int]
        :type original: int
        :rtype: int
        """
        found = set(nums)
        n = original

        while(n in found):
            n =n*2

        return n
        

File: 2154-keep-multiplying-found-values-by-two/2154-keep-multiplying-found-values-by-two.java
class Solution {
    public int findFinalValue(int[] nums, int original) {
        Set<Integer> found = new HashSet<>();

        for (int n:nums){
            found.add(n);
        }

        while(found.contains(original)){
            original = original*2;
        }

        return original;
    }
}